common/src/main/java/com/wynntils/features/ui/WynncraftButtonFeature.java

```diff
@@ -53,29 +58,44 @@ public void onTitleScreenInit(TitleScreenInitEvent.Post e) {
     public void onTitleScreenInit(ScreenInitEvent e) {
         if (!(e.getScreen() instanceof TitleScreen titleScreen)) return;
 
-        addWynncraftButton(titleScreen);
+        if (!hasInitBootPased && autoConnect.get()) {
+            ServerData wynncraftServer = getWynncraftServer();
+            connectToServer(wynncraftServer);
+        } else {
+            addWynncraftButton(titleScreen);
+        }
+        hasInitBootPased = true;
```

Comment:
Refactor this into:

```
if firstboot and autoconnect:
then set firstboot to false;
         connecttoserver;
         return

addwynncraftbutton
```

Reply:
i see now i get what DonkeyBlaster meant by inverting the firstTitleScreenInit!